lib/core/client.js

```diff
@@ -909,7 +909,10 @@ function _resume (client, sync) {
       request.body = null
     }
 
-    if (client.running && util.isStream(request.body)) {
+    if (client.running &&
+      !util.isBuffer(request.body) &&
+      typeof request.body !== 'string' &&
+      (util.isStream(request.body) || util.isIterable(request.body))) {
```

Comment:
This looks wrong, it should just be `if (client.running && (isStream(request.body) || isIterable(request.body)))`